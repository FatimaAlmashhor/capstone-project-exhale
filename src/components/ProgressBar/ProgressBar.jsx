import React from 'react';
import PropTypes from 'prop-types';

export default function ProgressBar({ now, max, showLabel }) {
  return (
    <div className="relative pt-1">
      {showLabel && <p className="text-textColor-primary">{`${now}\\${max}`}</p>}
      <div className="overflow-hidden  h-2 mb-4 text-xs flex rounded bg-blue-200">
        <div
          style={{ width: `${7 * now}%` }}
          className="shadow-none  flex flex-col text-center whitespace-nowrap text-white justify-center  bg-yellow-500 transition-all duration-200"
        />
      </div>
    </div>
  );
}

ProgressBar.propTypes = {
  now: PropTypes.number,
  max: PropTypes.number,
  showLabel: PropTypes.bool,
};

ProgressBar.defaultProps = {
  now: 0,
  max: 100,
  showLabel: false,
};
