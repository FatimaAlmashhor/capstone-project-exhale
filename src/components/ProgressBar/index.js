export { default } from './ProgressBar';
import React from 'react';
import renderer from 'react-test-renderer';
import Progress from './ProgressBar';

it('renders correctly', () => {
  const tree = renderer
    .create(<Progress />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});