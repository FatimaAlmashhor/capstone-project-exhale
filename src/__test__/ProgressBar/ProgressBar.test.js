import React from 'react';
import renderer from 'react-test-renderer';
import ProgressBar from '../../components/ProgressBar/ProgressBar';

it('renders correctly', () => {
  const tree = renderer.create(<ProgressBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
