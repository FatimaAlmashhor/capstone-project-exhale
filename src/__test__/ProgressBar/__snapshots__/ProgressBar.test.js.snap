// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative pt-1"
>
  <div
    className="overflow-hidden  h-2 mb-4 text-xs flex rounded bg-blue-200"
  >
    <div
      className="shadow-none  flex flex-col text-center whitespace-nowrap text-white justify-center  bg-yellow-500 transition-all duration-200"
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;
